lumicks/pylake/kymo.py

```diff
@@ -729,3 +729,81 @@ def line_timestamp_ranges_factory(_, exclude=bool):
     kymo._line_timestamp_ranges_factory = line_timestamp_ranges_factory
 
     return kymo
+
+
+def _kymo_from_correlated_stack(
+    corrstack, width=1, pixel_size_um=None, name="", reduce=np.mean
+) -> Kymo:
+    """Generate a `Kymo` instance from a correlated stack.
+
+    Parameters
+    ----------
+    corrstack : CorrelatedStack
+        An instance of a CorrelatedStack. The frame rate and the exposure time of the images need
+        to be constant and `corrstack` needs to have a tether. The data for the kymograph will be
+        taken along the line of the tether, including the pixels of the ends of the tether.
+    width : int
+        The width of the line in pixels the data for the kymograph will be taken. The width will
+        be rounded to the next biggest non negative odd number of pixels. The pixel values will be
+        reduced to a one pixel line given by :func:`reduce`.
```

Comment:
I used `width` by purpose, as this seemed the most intuitive way to me how someone calling the function would think of defining the line width the data for the kymo is taken from. `halfwidth` would imply the final `width` of the line would consist of an even number of lines. Probably, I could use something `adjacent_lines` or `extra_lines` instead of halfwidth?

Reply:
My main gripe is that by specifying the full width you either have to silently round it (like you do now), which could lead to people not getting what they expect. Silently doing something is often a source of errors. Or you raise or warn which also doesn't seem that nice for user experience.

In `KymoLine.sample_from_image()` we use `num_pixels`, but I agree that this is also not the most descriptive name.
```
        num_pixels : int
            Number of pixels in either direction to include in the sample
```

Reply:
Replace `width` with `adjacent_lines`.